### PR TITLE
[BugFix] Rewritten aggregate projection should not refer other column references in  the same projection mapping

### PR DESCRIPTION
**Pull request by LiShuMing**
Fixes #issue

It's not safe for column ref to refer other column refs in the same projection mapping.

```
mysql> explain select user_id, round(percentile_approx(tag_id, 0)) x from user_tags group by user_id order by user_id;
+------------------------------------------------------------------------------+
| Explain String                                                               |
+------------------------------------------------------------------------------+
| PLAN FRAGMENT 0                                                              |
|  OUTPUT EXPRS:2: user_id | 6: round                                          |
|   PARTITION: UNPARTITIONED                                                   |
|                                                                              |
|   RESULT SINK                                                                |
|                                                                              |
|   4:MERGING-EXCHANGE                                                         |
|                                                                              |
| PLAN FRAGMENT 1                                                              |
|  OUTPUT EXPRS:                                                               |
|   PARTITION: RANDOM                                                          |
|                                                                              |
|   STREAM DATA SINK                                                           |
|     EXCHANGE ID: 04                                                          |
|     UNPARTITIONED                                                            |
|                                                                              |
|   3:SORT                                                                     |
|   |  order by: <slot 2> 2: user_id ASC                                       |
|   |  offset: 0                                                               |
|   |                                                                          |
|   2:Project                                                                  |
|   |  <slot 2> : 9: user_id                                                   |
|   |  <slot 5> : percentile_approx_raw(13: percentile_union, 0.0)             |
|   |  <slot 6> : round(5: percentile_approx)                                  |
|   |                                                                          |
|   1:AGGREGATE (update finalize)                                              |
|   |  output: percentile_union(11: percentile_union(percentile_hash(tag_id))) |
|   |  group by: 9: user_id                                                    |
|   |                                                                          |
|   0:OlapScanNode                                                             |
|      TABLE: user_tags_percential_mv2                                         |
|      PREAGGREGATION: ON                                                      |
|      partitions=1/1                                                          |
|      rollup: user_tags_percential_mv2                                        |
|      tabletRatio=3/3                                                         |
|      tabletList=15079,15081,15083                                            |
|      cardinality=1                                                           |
|      avgRowSize=1048580.0                                                    |
+------------------------------------------------------------------------------+
38 rows in set (0.01 sec)
```

After this PR:
```
mysql> explain select user_id, round(percentile_approx(tag_id, 0)) x from user_tags group by user_id order by user_id;
+-----------------------------------------------------------------------------+
| Explain String                                                              |
+-----------------------------------------------------------------------------+
| PLAN FRAGMENT 0                                                             |
|  OUTPUT EXPRS:2: user_id | 6: round                                         |
|   PARTITION: UNPARTITIONED                                                  |
|                                                                             |
|   RESULT SINK                                                               |
|                                                                             |
|   4:MERGING-EXCHANGE                                                        |
|                                                                             |
| PLAN FRAGMENT 1                                                             |
|  OUTPUT EXPRS:                                                              |
|   PARTITION: RANDOM                                                         |
|                                                                             |
|   STREAM DATA SINK                                                          |
|     EXCHANGE ID: 04                                                         |
|     UNPARTITIONED                                                           |
|                                                                             |
|   3:SORT                                                                    |
|   |  order by: <slot 2> 2: user_id ASC                                      |
|   |  offset: 0                                                              |
|   |                                                                         |
|   2:Project                                                                 |
|   |  <slot 2> : 7: user_id                                                  |
|   |  <slot 6> : round(percentile_approx_raw(11: percentile_union, 0.0))     |
|   |                                                                         |
|   1:AGGREGATE (update finalize)                                             |
|   |  output: percentile_union(9: percentile_union(percentile_hash(tag_id))) |
|   |  group by: 7: user_id                                                   |
|   |                                                                         |
|   0:OlapScanNode                                                            |
|      TABLE: user_tags_percential_mv2                                        |
|      PREAGGREGATION: ON                                                     |
|      partitions=1/1                                                         |
|      rollup: user_tags_percential_mv2                                       |
|      tabletRatio=3/3                                                        |
|      tabletList=29715,29717,29719                                           |
|      cardinality=3                                                          |
|      avgRowSize=1048580.0                                                   |
+-----------------------------------------------------------------------------+
37 rows in set (0.01 sec)
```
## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java [hunk 1]
```diff
@@ -681,27 +681,36 @@ private LogicalAggregationOperator rewriteAggregationOperatorByRules(
...
-                newProjection.put(aggEntry.getKey(), aggEntry.getKey());
+                defaultProjection.put(aggEntry.getKey(), aggEntry.getKey());
                 newColumnRefToAggFuncMap.put(aggEntry.getKey(), aggEntry.getValue());
             }
         }
         // Copy group by keys as projection
-        aggregationOperator.getGroupingKeys().forEach(c -> newProjection.put(c, c));
+        aggregationOperator.getGroupingKeys().forEach(c -> defaultProjection.put(c, c));
+        // Make a new logical agg with new projections.
+        LogicalAggregationOperator newAggOp =
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java:698]: better to use Builder

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 13 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|13|13|100.00%|[]|

**Review by ABingHuang — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|8|8|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|10|10|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**